bootkit/main.go

```diff
@@ -146,8 +149,25 @@ func main() {
 
 	fmt.Printf("Pulling image [%s]", imageName)
 
-	out, err := cli.ImagePull(ctx, imageName, pullOpts)
-	if err != nil {
+	// TODO: Ideally if this function becomes a loop that runs forever and keeps retrying
+	// anything that failed, this retry would not be needed. For now, this addresses the specific
+	// race condition case of when the linuxkit network or dns is in the process of, but not quite
+	// fully set up yet.
+
+	failedImagePull := true
+
+	var out io.ReadCloser
+	for i := 0; i < imagePullRetryAttempts; i++ {
+		out, err = cli.ImagePull(ctx, imageName, pullOpts)
+		if err == nil {
+			failedImagePull = false
+			break
+		}
+		fmt.Printf("Error pulling image [%s] [%v]. Retrying after %d seconds...\n", imageName, err, retrySleepSeconds)
+		time.Sleep(time.Second * retrySleepSeconds)
+	}
+
+	if failedImagePull {
```

Comment:
I've got a small issue with plain-old-simple retries vs something smarter, can you rework with some exponential backoff please? I was just messing around with one that happens to be an indirect dep of tink and boots already and it even reads pretty nice to me: https://go.dev/play/p/9fbv6A9zQmq